src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(creature) {
+    if (creature instanceof Herbivore && !creature.hidden) {
+      creature.health -= 50;
+
+      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```

Comment:
you iterate over this array even if no creature died. this is not very efficient

Reply:
Oh! yah I get it. How about now?